### ISSUE TITLE
AgBot gets wrong token in all-in-one deployment.

### ISSUE THREAD

**Issue by TheMosquito** [open-horizon/devops#39]
I deployed an all-in-one, but no agreements ever formed, even though node registration succeeded. Looking at the agbot logs I saw lots of credential errors.

Looking inside the container, we discovered that the `/etc/horizon/anax.json` file had somehow gotten the wrong token! Editing this file to contain the correct token, and then stopping and starting the container fixed the problem. Agreements with registered nodes now happen as expected.

I noticed that this bind happens for the agbot container:

```
"Binds": [
                "/etc/horizon/agbot.json:/etc/horizon/anax.json.tmpl:ro"
...
```

Notice the `.tmpl` suffix. This file contains the correct token and is identical to the bound host file (as expected since it it s readonly bind). So I assume that the AgBot is supposed to copy this bound file into `anax.json` (without the `.tmpl` suffix) but for some reason it didn't just copy it; it edited the token, which broke the agbot.

**Comment by bmpotter** [comment]
@TheMosquito did this error occur on ubuntu or macos?

**Comment by TheMosquito** [comment]
@bmpotter  This was on ubuntu 18.04, running on an Intel NUC i7/16GB.

**Comment by bmpotter** [comment]
@TheMosquito hmmm, i have no idea how to debug this, because it works fine for me. Is this repeatable?

**Comment by bmpotter** [comment]
I figured out the problem: by default it generates random pw/tokens so the user doesn’t have to specify any args if they don’t want to (this can be overridden). So on the 2nd run, the agbot token was being changed in the exchange, but the running agbot wasn’t picking it up. In the fall timeframe we always restarted the agbot because of that bug where it wasn’t picking up the org. But i removed that recently, because the agbot bug was fixed. And that’s when this no-agreement problem started showing up on rerun. I fixed it such that it now restarts the agbot if the token has changed.
